src/dns.c

```diff
@@ -965,22 +965,26 @@ void dns_update (dns_t *dns, const struct pcap_pkthdr *header, const void *start
  * \return none
  */
 void dns_print_json (const dns_t *dns1, const dns_t *dns2, zfile f) {
-    unsigned int count;
-    char * const *twin_dns_name = NULL;
-    const unsigned short *twin_pkt_len = NULL;
+    unsigned int count = 0;
   
-    count = dns1->pkt_count > MAX_NUM_DNS_PKT ? MAX_NUM_DNS_PKT : dns1->pkt_count;
+    if (dns1) {
+        count = dns1->pkt_count > MAX_NUM_DNS_PKT ? MAX_NUM_DNS_PKT : dns1->pkt_count;
+    }
+
```

Comment:
Shouldn't dns2 below have the same checks for MAX_NUM_DNS_PKT ?

Reply:
no dns1 was checked for max and dns2 uses dns1 as its max

Reply:
Seems like you want dns2 check in dns1 block in case dns1 is null and dns2 depends on dns1

Reply:
if dns1 is NULL, then count stays at 0. If dns2 is not null and count is bigger than 0, then count stays at 0 because it uses prior count as the maximum. Basically if dns1 is null, the routine will exit because count will stay 0.